kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/QueryDataServiceBase.java

```diff
@@ -158,23 +158,17 @@ public Object queryFiltered(String queryName, String mapper, Integer page, Integ
 
     public Object queryFilteredWithBuilder(String queryName, String mapper, String builder, Integer page, Integer pageSize, String payload, String marshallingType) {
         Map<String, String> columnMapping = null;
-        QueryContext queryContext = buildQueryContext(page, pageSize);
-
-
+        AdvancedQueryContext queryContext = new AdvancedQueryContext(buildQueryContext(page, pageSize));
         Map<String, Object> queryParameters = new HashMap<String, Object>();
+
         if (payload != null && !payload.isEmpty()) {
             logger.debug("About to unmarshal query params from payload: '{}'", payload);
             queryParameters = marshallerHelper.unmarshal(payload, marshallingType, Map.class);
+            String orderByClause = (String) queryParameters.remove(KieServerConstants.QUERY_ORDER_BY_CLAUSE);
+            columnMapping = (Map<String, String>) queryParameters.remove(KieServerConstants.QUERY_COLUMN_MAPPING);
 
-            String orderBy = (String) queryParameters.remove(KieServerConstants.QUERY_ORDER_BY);
-            Boolean ascending = (Boolean) queryParameters.remove(KieServerConstants.QUERY_ASCENDING);
-            columnMapping = (Map<String, String> )queryParameters.remove(KieServerConstants.QUERY_COLUMN_MAPPING);
-
-            if (orderBy != null) {
-                queryContext.setOrderBy(orderBy);
-            }
-            if (ascending != null) {
-                queryContext.setAscending(ascending);
```

Comment:
why do we remove this basic sorting capabilities? I'd say we should still support it and only add the orderByClause on top of it. Meaning if there is oderByClause given use that otherwise use sorting as it is now.